### PR TITLE
Monitoring: Add Prometheus Targets UI pages

### PR DESCRIPTION
**Pull request by kyoto**
Epic: https://issues.redhat.com/browse/MON-1079

FYI @cshinn, @thile01

screenshot-2

screenshot-3

### CODE REVIEW

**Comment by kyoto** [comment]
Adding `docs-approved` because the epic already has the `doc-ack` label.

/label docs-approved

File: frontend/public/components/monitoring/targets.tsx [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+          labelFilter="observe-target-labels"
+          labelPath="labels"
+          nameFilterPlaceholder={t('public~Search by endpoint or monitor...')}
+          reduxIDs={[REDUX_ID]}
+          rowFilters={rowFilters}
+          textFilter="observe-target-text"
+        />
+        <div className="row">
+          <div className="col-xs-12">
+            <Table
```
Comment (spadgett) [inline frontend/public/components/monitoring/targets.tsx:308]: Don't redo this, but in general new code should use the updated table components from #8829.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by kyoto** [comment]
Adding `px-approved` because the epic already has the `px-ack` label.
FYI @RickJWagner

/label px-approved

**Comment by lihongyan1** [comment]
The Name filter drop down with value Name&Label should be change as Endpoint&Monitor

**Comment by lihongyan1** [comment]
The Name filter drop down with value Name&Label should be change as Endpoint&Monitor

**Comment by lihongyan1** [comment]
Order by Monitor can't works well, Not sure why some targets with null Monitor

**Comment by lihongyan1** [comment]
It would be better if admin can filter targets by namespace/project

**Comment by kyoto** [comment]
Thanks @lihongyan1. I've made the following changes to address the issues you found.
- Changed the text filter to match against Endpoint and Namespace (instead of Endpoint and Monitor)
- Changed the text filter label to be "Text" instead of "Name"
- Fixed a bug that was causing Monitor to be null for some targets
- Removed sorting from the Monitor column since it was broken and will be a little tricky to implement

@spadgett In addition to the above changes, I also
- Started using React context for the services watcher and service monitors watcher. This fixes a problem where the watchers would be destroyed and recreated when navigating between the list page and details page and also sometimes when scrolling through the list. This change noticeably improve performance.
- Memoized the `Health` component, since it only has 2 states (`up` and `down`)

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by kyoto** [comment]
Rebased. Could you tag again please @spadgett?

File: frontend/public/components/monitoring/targets.tsx [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+    ({ metadata }) =>
+      metadata.name === target?.labels?.service && metadata.namespace === target?.labels?.namespace,
+  );
+
+  // Now find the service monitor that corresponds to the service
+  const monitor = _.find(
+    monitors,
+    ({ spec }) =>
+      service &&
+      _.isMatch(service.metadata.labels, spec.selector?.matchLabels) &&
```
Comment (spadgett) [inline frontend/public/components/monitoring/targets.tsx:55]: I missed this previously, but what if the service uses `matchExpressions`?

I'd use `LabelSelector` from label-selector.js to handle `matchExpressions`.

**Comment by kyoto** [comment]
Thanks @spadgett. I've updated the PR to use `LabelSelector`.

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**